onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -797,10 +797,6 @@ Status QnnBackendManager::SetupBackend(const logging::Logger& logger,
                                        bool load_from_cached_context,
                                        bool need_load_system_lib) {
   std::lock_guard<std::recursive_mutex> lock(logger_recursive_mutex_);
-  if (backend_setup_completed_) {
```

Comment:
This SetupBackend will be invoked twice in GetCapability. We use this flag to avoid the 2nd call.

Reply:
I would recommend has a extra flag to force release resources void ReleaseResources(bool force_release = false)

void QnnBackendManager::ReleaseResources(bool force_release) {
  if (!backend_setup_completed_ && !force_release) {
    return;
  }

Reply:
I've put `backend_setup_completed_` back but removed the check in `ReleaseResources` so it's still clear that this needs to be safe to be called more than once as this is the same as calling it on a partially complete initialisation

Reply:
Sorry didn't see your message before making my change. I think it might be better to remove the check to make it clear it needs to be safe to run multiple times?